### PR TITLE
DxDiagram: default item properties

### PR DESCRIPTION
**Pull request by LanaDX**

### CODE REVIEW

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineEnd.md [hunk 1]
```diff
@@ -6,7 +6,12 @@ default: 'arrow'
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies an edge's default line end tip.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineEnd.md]:
```suggestion
Specifies the end of an edge's default line.
```

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineEnd.md [hunk 2]
```diff
@@ -6,7 +6,12 @@ default: 'arrow'
...
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+The **connectorLineEnd** option specifies a default line end tip that is applied to a connector if there is not another value provided via the [toLineEndExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/edges/#toLineEndExpr) option.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineEnd.md]: "default line end tip" Explain

Comment (dirk-pieterse) [reply api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineEnd.md]: "if there is not another value provided via the [toLineEndExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/edges/#toLineEndExpr) option" ->  if the [toLineEndExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/edges/#toLineEndExpr) option does not provide another value

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineStart.md [hunk 3]
```diff
@@ -6,7 +6,12 @@ default: 'none'
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies an edge's default line start tip.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+The **connectorLineStart** option specifies a default line start tip that is applied to a connector if there is not another value provided via the [fromLineEndExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/edges/#fromLineEndExpr) option.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineStart.md]: Why "a default"?

Comment (dirk-pieterse) [reply api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineStart.md]: "if there is not another value provided via the [fromLineEndExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/edges/#fromLineEndExpr) option" This is a strange way to say "if the **fromLineEndExpr** option does not provide another value"

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineType.md [hunk 4]
```diff
@@ -6,7 +6,12 @@ default: 'orthogonal'
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies an edge's default line type.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+The **connectorLineType** option specifies a default line type that is applied to a connector if there is not another value provided via the [lineTypeExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/edges/#lineTypeExpr) option.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineType.md]: Same here

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/defaultItemProperties.md [hunk 5]
```diff
@@ -5,7 +5,23 @@ default: {}
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Configures default item properties.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+Default property values are applied to newly created (via the UI or during data binding) items if no other values are provided.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/defaultItemProperties.md]:
```suggestion
Default property values are applied to newly created items if no other values are provided.
```

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/style.md [hunk 6]
```diff
@@ -4,7 +4,12 @@ type: Object
 ---
 ---
 ##### shortDescription
-<!-- Description goes here -->
+Specifies a default item style.
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+The **style** option specifies a default style that is applied to a shape/connector if there is not another value provided via the [nodes.styleExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/nodes/#styleExpr)/[edges.styleExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/edges/#styleExpr) option.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/style.md]:
```suggestion
The **style** option specifies the default style applied to a shape/connector if the [nodes.styleExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/nodes/#styleExpr)/[edges.styleExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/edges/#styleExpr) option does not provide another value. 
```

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/edges/styleExpr.md [hunk 7]
```diff
@@ -11,7 +11,9 @@ Specifies the name of a data source field or an expression that provides an edge
 The current edge's data object.
 
 ---
-The style settings must be specified as CSS rules presented in JSON format, for instance `{ "stroke": "#444444" }`.
+A data source field specified by this option must contain inline style declarations in string format, for instance `"fill: #d9d9d9; stroke: #999999"`.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/edges/styleExpr.md]:
```suggestion
A data source field specified by this option should contain inline style declarations in string format, for instance `"fill: #d9d9d9; stroke: #999999"`. 
```

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/textStyle.md [hunk 8]
```diff
@@ -4,7 +4,12 @@ type: Object
...
 
 ---
-<!-- Description goes here -->
\ No newline at end of file
+The **textStyle** option specifies a default text style that is applied to a shape/connector if there is not another value provided via the [nodes.textStyleExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/nodes/#textStyleExpr)/[edges.textStyleExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/edges/#textStyleExpr) option.
+
+The style settings can be specified in the following ways:
+
+- as inline style declaration, for instance `"font-weight: bold; text-decoration: underline"`,
+- as CSS rules presented in JSON format, , for instance `{ "font-weight": "bold", "text-decoration": "underline" }`.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/textStyle.md]:
```suggestion
- as CSS rules in JSON format, for instance `{ "font-weight": "bold", "text-decoration": "underline" }`.
```

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/nodes/textStyleExpr.md [hunk 9]
```diff
@@ -11,7 +11,25 @@ Specifies the name of a data source field or an expression that provides a node'
 The current node's data object.
 
 ---
-The style settings must be specified as CSS rules presented in JSON format, for instance `{ "font-weight": "bold", "text-decoration": "underline" }`.
+A data source field specified by this option must contain inline style declarations in string format, for instance `"font-weight: bold; text-decoration: underline"`. 
+
+If you provide a function for the **textStyleExpr** option, the function can return style settings as CSS rules presented in JSON format as well.
```
Comment (dirk-pieterse) [inline api-reference/10 UI Widgets/dxDiagram/1 Configuration/nodes/textStyleExpr.md]:
```suggestion
If you provide a function for the **textStyleExpr** option, the function can return style settings as CSS rules in JSON format.
```

**Review by dirk-pieterse — APPROVED** [review]

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineEnd.md (hunk 1 above)
Comment (LanaDX) [reply api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineEnd.md:9]: Perhaps
```suggestion
Specifies the tip of a connector's end point.
```

File: api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineEnd.md (hunk 2 above)
Comment (LanaDX) [reply api-reference/10 UI Widgets/dxDiagram/1 Configuration/defaultItemProperties/connectorLineEnd.md:12]:
```suggestion
The **connectorLineEnd** option specifies a tip that is displayed at a connector's end point if the [toLineEndExpr](/Documentation/ApiReference/UI_Widgets/dxDiagram/Configuration/edges/#toLineEndExpr) option does not provide another value.
```

**Pull request merged**